shipyard_blueprints/s3/shipyard_s3/s3.py

```diff
@@ -1,39 +1,177 @@
+from shipyard_s3.utils.exceptions import (
+    DownloadError,
+    InvalidCredentials,
+    InvalidRegion,
+    MoveError,
+    RemoveError,
+    UploadError,
+)
+from shipyard_s3.utils import utils
+
+
+logger = ShipyardLogger.get_logger()
 
 
 class S3Client(CloudStorage):
     def __init__(
-        self, aws_access_key: str, aws_secret_access_key: str, region: str = None
+        self,
+        aws_access_key: str,
+        aws_secret_access_key: str,
+        region: Optional[str] = None,
     ) -> None:
         self.aws_access_key = aws_access_key
         self.aws_secret_access_key = aws_secret_access_key
         self.region = region
-        super().__init__(
-            aws_access_key=aws_access_key,
-            aws_secret_access_key=aws_secret_access_key,
-            region=region,
-        )
+
+    @property
+    def s3_conn(self):
+        return self.connect()
 
     def connect(self):
-        client = boto3.client(
-            "s3",
-            region_name=self.region,
-            aws_access_key_id=self.aws_access_key,
-            aws_secret_access_key=self.aws_secret_access_key,
-            config=None,
-        )
-        self.logger.info("Successfully connected to S3")
-        return client
+        try:
+            logger.debug("Establishing connection with S3")
+            client = boto3.client(
+                "s3",
+                region_name=self.region,
+                aws_access_key_id=self.aws_access_key,
+                aws_secret_access_key=self.aws_secret_access_key,
+                config=None,
+            )
+        except Exception as e:
+            logger.error(str(e))
+            raise InvalidCredentials
+        else:
+            logger.debug(
+                "Successfully established connection, initializing connection property"
+            )
+            return client
+
+    def move(self, src_bucket: str, dest_bucket: str, src_path: str, dest_path: str):
+        """Move or rename an object within an S3 bucket
+
+        Args:
+            src_bucket: The source bucket
+            dest_bucket: The destination bucket (can be the same as the src_bucket) if you are not moving objects between buckets
+            src_path: The path of where the source object is
+            dest_path: The path to where the source objects should be moved to
+
+        Raises:
+            MoveError:
+        """
+        try:
+            # create a source dictionary that specifies bucket name and key name of the object to be copied
+            copy_source = {"Bucket": src_bucket, "Key": src_path}
+            # move the object(s)
+            self.s3_conn.copy_object(
+                Bucket=dest_bucket, CopySource=copy_source, Key=dest_path
+            )
+            # delete the original
+            self.s3_conn.delete_object(Bucket=src_bucket, Key=src_path)
+
+        except Exception as e:
+            if "IllegalLocationConstraintException" in str(e):
```

Comment:
this shouldn't be nested.

if should be more like
``` 
from boto3.exceptions import IllegalLocationConstraintException
try:
    ...
except IllegalLocationConstraintException as err:
     raise InvalidRegion(region) from err
except Exception as err:
    raise MoveError('An error occurred when attempting to move file. Message from s3 server: {err})
```

Reply:
The problem is that is not an actual exception, it is just present in the message [](edited-with-Axolo)